cli/join.go

```diff
@@ -15,6 +17,26 @@ func checkAddrFormat(addr string) bool {
 	return m
 }
 
+func checkAddrValidity(addr string) bool {
+	newip := strings.Split(addr, ":")
+	// Handle loopback interface as a special case
+	if strings.EqualFold(newip[0], "localhost") {
+		return true
+	}
+	// Compare host's network interface addresses with requested address
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
if err, should we `return false` just after logging fatal?
I think `addrs` is not valid after `err` is not `nil`.